src/cast.js

```diff
@@ -12,16 +12,16 @@ export function cast(input, type) {
   let outputArray;
   switch (type) {
     case 'int8':
-      outputArray = new Int8Array(input.data);
+      outputArray = new Int8Array(Array.from(input.data, (num) => (Math.round(num))));
       break;
     case 'uint8':
-      outputArray = new Uint8Array(input.data);
+      outputArray = new Uint8Array(Array.from(input.data, (num) => (Math.round(num))));
       break;
     case 'int32':
-      outputArray = new Int32Array(input.data);
+      outputArray = new Int32Array(Array.from(input.data, (num) => (Math.round(num))));
       break;
     case 'uint32':
-      outputArray = new Uint32Array(input.data);
+      outputArray = new Uint32Array(Array.from(input.data, (num) => (Math.round(num))));
```

Comment:
Truncation toward zero is the expected casting behavior for float to int, not rounding to nearest. That's consistent with existing ML libraries I know of. e.g. `float32` 3.7 -> `uint32` 3. Isn't the default behavior of `Uint32Array` already the expected truncation toward zero behavior?

Reply:
I'm sorry for the misunderstanding. I've revised the code :)